### PR TITLE
Add unit tests for availability helpers #830

### LINKED ISSUE #830: Add direct unit tests for `app/tools/utils/availability.py`
## Why this is a good first issue
very small helper module, clear expected behavior, no network calls.

## Likely files
- `app/tools/utils/availability.py`
- `tests/tools/`

## Acceptance criteria
- add tests for `eks_available_or_backend`
- add tests for `datadog_available_or_backend`
- add tests for `cloudwatch_is_available`
- cover both verified credentials and injected `_backend` cases

## First-time contributors
This issue is intentionally scoped to be approachable for first-time contributors: one focused change, clear files, and low regression risk.

### PR DESCRIPTION
**Pull request by Dipxssi**
Fixes #830 

 

#### Describe the changes you have made in this PR -

I have added a new test file `tests/tools/test_availability.py` that provides direct unit test coverage for the helper functions in `app/tools/utils/availability.py`. This includes:
*   Full coverage for `eks_available_or_backend` and `datadog_available_or_backend`, including edge cases for missing keys, verified credentials, and injected `_backend` objects (both truthy and falsy like `None`).
*   Tests for `cloudwatch_is_available` to verify its truthiness-based availability logic.

### Screenshots of the test cases  -
Screenshot 2026-04-24 235406

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

The goal was to provide unit tests for a small but critical utility module. I chose a class-based structure in `pytest` to group tests by function, which matches the project's existing testing style. 

The implementation focuses on verifying the dictionary-based lookups in the `sources` object. I considered whether to change the CloudWatch logic to "key presence" vs "truthiness" and decided to mirror the existing implementation's truthiness check while adding tests for edge cases like `_backend: None`, which ensures the tools don't accidentally report as "available" when a mock backend is explicitly disabled.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

### CODE REVIEW

**Comment by Devesh36** [comment]
Hey @Dipxssi could you please solve the greptile comments . Thankss !!

**Comment by Dipxssi** [comment]
@Devesh36 , yes I will do it .

File: tests/tools/test_availability.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        sources = {"eks": {"_backend": object()}}
+        assert eks_available_or_backend(sources) is True
+
+    def test_eks_not_available(self) -> None:
+        sources = {"eks": {"connection_verified": False}}
+        assert eks_available_or_backend(sources) is False
+
+    def test_eks_backend_none(self) -> None:
+        sources = {"eks": {"_backend": None}}
+        assert eks_available_or_backend(sources) is False
```
[the remark this answers is not in the document]
Comment (Dipxssi) [reply tests/tools/test_availability.py:35]: Added the combined-condition edge cases for both EKS and Datadog in the latest commit.

**Review by VaibhavUpreti — APPROVED** [review]
awesome @Dipxssi !
